src/betterproto/__init__.py

```diff
@@ -1022,6 +1022,29 @@ def FromString(cls: Type[T], data: bytes) -> T:
         """
         return cls().parse(data)
 
+    # For compatibility with Google protobuf official implementation.
+    def ParseFromString(self, data: bytes) -> T:
+        """
+        Parse the binary encoded Protobuf into this message instance. This
+        returns the instance itself and is therefore assignable and chainable.
+
+        .. note::
+            This is a method for compatibility with other libraries,
+            you should really use :meth:`parse`.
+
+
+        Parameters
+        -----------
+        data: :class:`bytes`
+            The data to parse the protobuf from.
+
+        Returns
+        --------
+        :class:`Message`
+            The initialized message.
+        """
+        return self.parse(data)
```

Comment:
Isn't this slightly different from the offical implementation which clears any currently set values? https://github.com/protocolbuffers/protobuf/blob/2a2a9b6e64e0cda49b0e5377cad58c790087e1c9/python/google/protobuf/message.py#L193-L199

Reply:
This indeed seems to be the case.

I've added a simple example which shows the difference in result from `ParseFromString` with the one from this PR:

```protobuf
syntax = "proto3";

package some.package.name;

message Cat {
  int64 age = 1;
  int64 lives = 2;
}
```

```python
def ParseFromString(self, data: bytes):
    return self.parse(data)

if __name__ == '__main__':
    # Create bytes
    cat = Cat(
        age=5,
        # lives=9,
    )

    data = cat.SerializeToString()

    # Dynamically add ParseFromString to the Cat class
    Cat.ParseFromString = ParseFromString

    # ParseFromString for the betterproto cat and the pb2 cat
    cat_betterproto = Cat(age=3, lives=8)
    cat_pb2 = Cat_pb2(age=3, lives=8)

    cat_betterproto.ParseFromString(data)
    cat_pb2.ParseFromString(data)

    print(cat_betterproto)  # Cat(age=5, lives=8), lives are not cleared
    print(cat_pb2)  # age: 5, lives are cleared
    assert cat_pb2.lives == 0
```